### PR TITLE
Bug 1129115 - Display error page when detecting a Elasticsearch error.

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

File: kuma/search/utils.py [hunk 1]
```diff
@@ -1,4 +1,12 @@
+import elasticsearch
 from urlobject import URLObject
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import exception_handler
+from tower import ugettext_lazy as _
+
+SEARCH_DOWN_DETAIL = _('Search is temporarily unavailable. '
+                       'Please try again in a few minutes')
```
Comment (robhudson) [inline kuma/search/utils.py]: The original had a period. Best to not change it so l10n is still in place.

File: settings.py [hunk 2]
```diff
@@ -1267,3 +1267,7 @@ def get_user_url(user):
 BANISH_ABUSE_THRESHOLD = sys.maxint # TODO: https://bugzil.la/1122658
 BANISH_USE_HTTP_X_FORWARDED_FOR = True
 BANISH_MESSAGE = _("This connection has been banned for suspicious activity.")
+
+REST_FRAMEWORK = {
+    'EXCEPTION_HANDLER': 'kuma.search.utils.search_exception_handler'
+}
```
Comment (robhudson) [inline settings.py]: Is search the only place DRF is used?

File: kuma/search/utils.py [hunk 3]
```diff
@@ -47,3 +55,16 @@ def clean_params(self, params):
...
+
+def search_exception_handler(exc):
+    # Call REST framework's default exception handler first,
+    # to get the standard error response.
+    response = exception_handler(exc)
+
+    if (response is None and
+            isinstance(exc, elasticsearch.ElasticsearchException)):
+        return Response({'error': SEARCH_DOWN_DETAIL},
+                        status=status.HTTP_200_OK)
```
Comment (robhudson) [inline kuma/search/utils.py]: Maybe a note here that 503 would be the correct response but due to the proxy, blah blah blah.

File: kuma/search/tests/test_views.py [hunk 4]
```diff
@@ -132,6 +132,39 @@ def test_allowed_methods(self):
...
+    def test_unhandled_exceptions(self):
+        class RealExceptionSearchView(SearchView):
+            filter_backends = ()
+
+            def list(self, *args, **kwargs):
+                raise ValueError
+
+        view = RealExceptionSearchView.as_view()
+        request = self.get_request('/en-US/search')
+        self.assertRaises(ValueError, view, request)
```
Comment (robhudson) [inline kuma/search/tests/test_views.py]: Nice tests.

File: settings.py (hunk 2 above)
Comment (jezdez) [inline settings.py]: Yeah, right now at least. I'll add a TODO to investigate once we use it more.

File: kuma/search/utils.py (hunk 3 above)
Comment (jezdez) [inline kuma/search/utils.py]: :+1:

File: kuma/search/tests/test_views.py (hunk 4 above)
Comment (jezdez) [inline kuma/search/tests/test_views.py]: ;)

File: kuma/search/utils.py (hunk 3 above)
Comment (robhudson) [inline kuma/search/utils.py]: Forgot to add the code comment here?

**Pull request merged**